homeassistant/components/template/light.py

```diff
@@ -371,6 +371,18 @@ async def async_turn_on(self, **kwargs):
                 kwargs[ATTR_COLOR_TEMP],
             )
             self._temperature = kwargs[ATTR_COLOR_TEMP]
+            if self._color_template is None:
+                self._color = None
```

Comment:
What's the reason for the check and reset?

Reply:
The check and reset is because template light has not yet been migrated to color mode, it still relies on compatibility code in the base light class to infer the color mode from the attributes.

That compatibility code will not work as intended if both `hs_color` and `color_temperature` are set (the `color_mode` will always be `hs`).

When template light is migrated to color_mode, the check and reset can be removed, and the `color_mode` will instead be optimistically set.